sympy/core/function.py

```diff
@@ -1200,9 +1177,20 @@ def __new__(cls, expr, *variables, **kwargs):
         # expression at all. Note, this cannot check non-symbols like
         # functions and Derivatives as those can be created by intermediate
         # derivatives.
-        if evaluate and all(isinstance(sc[0], Symbol) for sc in variable_count):
-            symbol_set = set(sc[0] for sc in variable_count if sc[1].is_positive)
-            if symbol_set.difference(expr.free_symbols):
+        if evaluate:
+            zero = False
+            free = expr.free_symbols
+            for v, c in variable_count:
+                vfree = v.free_symbols
+                if c.is_positive and vfree:
```

Comment:
I guess this was already there, but why is is_posistive needed here?

Reply:
A count can be positive, negative, 0 or symbolic (unknown). I am not sure what a negative power means -- antiderivative? I think the rationale here is we only know that differentiation wrt a quantity that has no symbol in common with the expr being differentiated if the count is positive. If the count is negative, I don't know what that means and if the count is 0 it doesn't matter. If the count is symbolic we don't know since it may be zero.

Reply:
Oh it's the count. Yes I remember this discussion now. It has to do with symbolic counts.